kernel/src/net/socket/util/options.rs

```diff
@@ -247,6 +255,14 @@ pub const MIN_RECVBUF: u32 = 2304;
 pub(in crate::net) trait GetSocketLevelOption {
     /// Returns whether the socket is in listening state.
     fn is_listening(&self) -> bool;
+    /// Returns the peer credentials.
+    fn peer_cred(&self) -> Option<CUserCred> {
+        None
+    }
+    /// Returns the peer groups.
+    fn peer_groups(&self) -> Result<Arc<[Gid]>> {
+        return_errno_with_message!(Errno::ENODATA, "the socket does not have peer groups");
+    }
```

Comment:
These options should be specific to UNIX sockets?

I think it's possible to match these socket options in `UnixStreamSocket::{get,set}_options`. So in `OptionSet`, we can just fail. After all, no information is stored inside the `OptionSet` so it cannot know the socket credentials or the groups.

Is is better?